### ISSUE TITLE
GPU Memory Requirements

### ISSUE THREAD

**Issue by yuriydzyuba** [MSP-UTD/MSP-Podcast_Challenge#3]
@ilucasgoncalves I was looking at the code wanted to know what GPU memory size was used during training for categorical SER. It looks like a single GPU was used, and I am assuming a GPU with over 20GB was needed given the large wavlm.

**Comment by 3loi** [comment]
We have access to multiple GPUs and have trained on all of them ranging from a single 24GB (RTX 3090) up to multiple 48GB GPUs (RTX 8000).  Regardless, if you wish to train with less VRAM, notice the "-accumulation_steps" argument, which allows you to run the forward pass multiple times (accumulation_steps times), before a backward step.

Upon running our code for a few iterations, here are the VRAM usage:
Batchsize = 32, accumulation = 1: VRAM ≈ 38GB
Batchsize = 32, accumulation = 4: VRAM ≈ 14GB

**Comment by ilucasgoncalves** [comment]
FYI: accumulation steps is used to set up [gradient accumulation](https://kozodoi.me/blog/20210219/gradient-accumulation).
